File: saml/src/main/java/com/zorroa/security/saml/MvcConfig.java
package com.zorroa.security.saml;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.method.support.HandlerMethodArgumentResolver;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;

import java.util.List;

@ComponentScan("com.zorroa.security.saml")
@Configuration
public class MvcConfig extends WebMvcConfigurerAdapter {

    @Autowired
    CurrentUserHandlerMethodArgumentResolver currentUserHandlerMethodArgumentResolver;

    @Override
    public void addArgumentResolvers(List<HandlerMethodArgumentResolver> argumentResolvers){
        argumentResolvers.add(currentUserHandlerMethodArgumentResolver);
    }

}

File: saml/src/main/kotlin/com/zorroa/security/saml/MvcConfig.kt
package com.zorroa.security.saml

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.web.method.support.HandlerMethodArgumentResolver
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@ComponentScan("com.zorroa.security.saml")
@Configuration
class MvcConfig : WebMvcConfigurer {

    @Autowired
    lateinit var currentUserHandlerMethodArgumentResolver: HandlerMethodArgumentResolver

    override fun addArgumentResolvers(argumentResolvers: MutableList<HandlerMethodArgumentResolver>) {
        argumentResolvers.add(currentUserHandlerMethodArgumentResolver)
    }

}


